lib/Dialect/FIRRTL/Transforms/SpecializeOption.cpp

```diff
@@ -27,40 +27,38 @@ namespace {
 struct SpecializeOptionPass
     : public circt::firrtl::impl::SpecializeOptionBase<SpecializeOptionPass> {
   using SpecializeOptionBase::numInstances;
-  using SpecializeOptionBase::select;
 
   void runOnOperation() override {
     auto circuit = getOperation();
-    if (select.empty()) {
-      markAllAnalysesPreserved();
-      return;
-    }
 
     DenseMap<StringAttr, OptionCaseOp> selected;
-    for (const auto &optionAndCase : select) {
-      size_t eq = optionAndCase.find("=");
-      if (eq == std::string::npos) {
-        mlir::emitError(circuit.getLoc(),
-                        "invalid option format: \"" + optionAndCase + '"');
-        return signalPassFailure();
-      }
+    if (auto choiceAttr = circuit.getSelectInstChoiceAttr()) {
+      for (auto attr : choiceAttr.getAsRange<StringAttr>()) {
+        const auto &optionAndCase = attr.getValue().str();
```

Comment:
Is a `const std::string &` safe to take for a `.str()` that isn't given a temporary?

Reply:
Thanks for spotting this. Fixed.